src/components/App.tsx

```diff
@@ -65,28 +65,28 @@ export default function App({
   }, [config])
   return (
     <Styles>
-      <div className={styles.app}>
-        <Header />
-        {(props as PageProps).uri === '/' && (
-          <Alert text={warning} state="info" />
-        )}
-        {isAccountInPurgatory && (
-          <Alert
-            title={purgatory.title}
-            badge={`Reason: ${accountPurgatory?.reason}`}
-            text={purgatory.description}
-            state="error"
-          />
-        )}
-        {client ? (
-          <ApolloProvider client={client}>
+      {client ? (
```

Comment:
not sure about the whole change in this file, what was the intention? This logic would prevent anything from being rendered unless the Apollo `client` is present?

Reply:
done that because i needed the Apollo client above the footer so i could use it in footer which includes MarketStats. i'm thinking about this...is there another way of doing this better?

Reply:
ah got it, didn't think of that. Current live implementation tries to make sure that header & footer are not re-rendered upon page changes, slo so that not every page change initializes new API requests happening in footer. So that would be the goal somehow.

Alternatively, since it's Gatsby, the correct place for providers wrapping the whole app would be in `helpers/wrapRootElement` so maybe we want to move this there. But think @mihaisc ran into issues with that back then for initial implementation